java/se/sics/mspsim/core/Chip.java

```diff
@@ -75,6 +76,11 @@ public Chip(String id, String name, MSP430Core cpu) {
     if (cpu != null) {
       logger = cpu.getLogger();
       cpu.addChip(this);
+    } else {
+      if (!(this instanceof MSP430Core thisCPU)) {
+        throw new IllegalArgumentException("Initializing Chip without an MSP430Core available");
+      }
+      logger = new DefaultEmulationLogger(thisCPU, System.out);
```

Comment:
What do you think about setting `cpu` to `this` here so the `Chip.cpu` variable is never set to null?

Reply:
Without having looked at any code, I think that is a good change, but that should be a separate PR. (This PR moves things around, but the previous invariants remain unchanged, and things are mostly initialized in the same order as before.)

Reply:
Sound reasonable.